crates/sui-core/src/signature_mpc/mod.rs

```diff
@@ -340,34 +349,84 @@ impl SignatureMPCAggregator {
             SignatureMPCMessageProtocols::Sign(m) => {
                 let mut state = sign_session_states.entry(session_id).or_insert_with(|| {
                     SignState::new(
+                        tiresias_key_share_decryption_key_share,
                         tiresias_public_parameters.clone(),
                         epoch,
                         party_id,
-                        parties,
+                        parties.clone(),
                         session_id,
                     )
                 });
 
                 let _ = state.insert_first_round(sender_party_id, m.clone());
 
+                if state.should_identify_malicious_actors() {
+                    if let Ok(SignRoundCompletion::MaliciousPartiesOutput(malicious_parties)) =
+                        identify_malicious(&state)
+                    {
+                        println!("Identified malicious parties: {:?}", malicious_parties);
+                    }
+                    return;
+                }
                 if let Some(r) = sign_session_rounds.get_mut(&session_id) {
                     if state.ready_for_complete_first_round(&r) {
                         drop(r);
-                        let state = state.clone();
                         Self::spawn_complete_sign_round(
                             epoch,
                             epoch_store.clone(),
                             party_id,
                             session_id,
                             session_ref,
-                            state,
                             sign_session_rounds.clone(),
                             sign_session_states.clone(),
                             submit.clone(),
                         );
                     }
                 }
             }
+            SignatureMPCMessageProtocols::SignProofs(
+                prover_party_id,
+                new_proofs,
+                message_indices,
+                involved_parties,
+            ) => {
```

Comment:
As you said, the party id is redundant - we removed it. Regarding the other parameter, we need them, and beside adhering to the current architecture of the code, which is far from perfect, I don't see a reason to remove them. What can we change so this will make more sense for you?